core/admin/mailu/internal/nginx.py

```diff
@@ -88,4 +91,18 @@ def get_server(protocol, authenticated=False):
             hostname, port = extract_host_port(app.config['AUTHSMTP_ADDRESS'], 10025)
         else:
             hostname, port = extract_host_port(app.config['SMTP_ADDRESS'], 25)
+    try:
+        # test if hostname is already resolved to an ip adddress
+        ipaddress.ip_address(hostname)
+    except:
+        # hostname is not an ip address - so we need to resolve it
+        hostname = resolve_hostname(hostname)
     return hostname, port
+
+@tenacity.retry(stop=tenacity.stop_after_attempt(100),
```

Comment:
I do not like this here, I would have prevered something more generic on the path to getting back to resolving things dynamically. But I guess it is a necessary evil toward it :)